### PR TITLE
DO NOT MERGE: CAP-46 hminimal-host-objects branch

### PR DESCRIPTION
**Pull request by graydon**
This is a draft PR which does not really represent a request-for-merge. It's a head I will be maintaining while working on CAP-46 with a minimal set of host objects and functions. I am opening a PR just to provide an explanation for the branch, and in case anyone wants to comment on it (eg. make suggestions about what else to cut out while working on CAP-46).

### CODE REVIEW

**Comment by graydon** [comment]
(I will also temporarily be pointing core at this branch for integration purposes, and deleting all debated host object type codes from core's XDR. I'd welcome any thoughts about how to manage the ongoing gap between XDR that's approved-and-merged into core and XDR that's still "host objects we're debating".)

File: stellar-contract-env-common/src/env.rs [hunk 1]
```diff
@@ -60,97 +60,17 @@ macro_rules! call_macro_with_all_host_functions {
             // argument list of comma-separated arg:type pairs
 
             mod context "x" {
-                {"$_", fn log_value(v:RawVal) -> RawVal }
-                {"$0", fn get_last_operation_result() -> RawVal }
-                {"$1", fn obj_cmp(a:RawVal, b:RawVal) -> i64 }
+                {"$_", fn obj_cmp(a:RawVal, b:RawVal) -> i64 }
```
Comment (jayz22) [inline stellar-contract-env-common/src/env.rs:63]: Should `a` and `b` also be `Object` type?

**Comment by jayz22** [comment]
@graydon Not sure if you were planning to commit some of these changes to the `main` branch, I've copied your changes to `obj_to_i64`, `obj_to_u64` into [this PR](https://github.com/stellar/rs-stellar-contract-env/pull/33). If you've already added them or planning to add them yourself, feel free to discard.

File: stellar-contract-env-common/src/env.rs (hunk 1 above)
Comment (graydon) [reply stellar-contract-env-common/src/env.rs:63]: Oh, yes, probbaly so. Good catch.

**Comment by graydon** [comment]
> Not sure if you were planning to commit some of these changes to the main branch, I've copied your changes to obj_to_i64, obj_to_u64 into https://github.com/stellar/rs-stellar-contract-env/pull/33

That's great, thanks!

**Comment by graydon** [comment]
I think this branch has outlived its usefulness. Closing.

**Pull request closed without merging**